lib/src/fsmonitor.rs

```diff
@@ -242,5 +254,60 @@ pub mod watchman {
                 Ok((clock, Some(paths)))
             }
         }
+
+        /// Return whether or not a trigger has been registered already.
+        #[instrument(skip(self))]
+        async fn is_trigger_registered(&self) -> Result<bool, Error> {
+            info!("Checking for an existing Watchman trigger...");
+            Ok(list_triggers(&self.client, &self.resolved_root)
+                .await
+                .map_err(Error::WatchmanTriggerError)?
+                .triggers
+                .iter()
+                .any(|t| t.name == "jj-background-monitor"))
+        }
+
+        /// Register trigger for changed files.
+        #[instrument(skip(self))]
+        async fn register_trigger(&self) -> Result<(), Error> {
+            // TODO: might be better to specify query options by caller, but we
+            // shouldn't expose the underlying watchman API too much.
+            info!("Registering Watchman trigger...");
+            register_trigger(
+                &self.client,
+                &self.resolved_root,
+                TriggerRequest {
+                    name: "jj-background-monitor".to_string(),
+                    command: vec!["jj".to_string(), "st".to_string()],
```

Comment:
nit: `jj st` do more CPU intensive work than snapshotting. I think `jj files -r 'root()'` is cheaper. We might want some debug command for this kind of operation, but it can be added later.

It might also be better to pass the executable name from CLI, but this can be addressed later.

Reply:
Done